api/v1alpha1/storageclassclaim_types.go

```diff
@@ -73,7 +73,9 @@ type StorageClassClaim struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   StorageClassClaimSpec   `json:"spec,omitempty"`
+	//+kubebuilder:validation:Required
+	//+kubebuilder:validation:XValidation:rule="oldSelf == self",message="spec is immutable"
```

Comment:
does this rule enforces all fields in the struct as well? like, after setting the spec, does `.spec.storageProfile` also becomes immutable?

Reply:
As part of similar work we did for storage profiles, we observed that it does enforce on the entire structure and sub structures. 

@leelavg If you are satisfied please resolve this thread

Reply:
Can be closed.